### ISSUE TITLE
CLI does not exit with error when AWS permissions missing

### ISSUE THREAD

**Issue by kairstenfay** [nextstrain/cli#74]
I noticed this bug when running the `rebuild-staging` script in the private `ncov-ingest` repo. See [this conversation](https://github.com/nextstrain/ncov-ingest/pull/31#issuecomment-611703873) for more context.

**Comment by tsibley** [comment]
Thanks, Kairsten. That's a private repo, though, so could you summarize context here a bit more?

**Comment by kairstenfay** [comment]
When I try to run the `ncov` build without AWS permissions, I get this output:
```
Traceback (most recent call last):
  File "/home/kairsten/miniconda3/envs/ncov-ingest/lib/python3.6/site-packages/nextstrain/cli/runner/aws_batch/s3.py", line 193, in bucket
    boto3.client("s3").head_bucket(Bucket = bucket.name)
  File "/home/kairsten/miniconda3/envs/ncov-ingest/lib/python3.6/site-packages/botocore/client.py", line 276, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/kairsten/miniconda3/envs/ncov-ingest/lib/python3.6/site-packages/botocore/client.py", line 586, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (403) when calling the HeadBucket operation: Forbidden

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kairsten/miniconda3/envs/ncov-ingest/bin/nextstrain", line 8, in <module>
    sys.exit(main())
  File "/home/kairsten/miniconda3/envs/ncov-ingest/lib/python3.6/site-packages/nextstrain/cli/__main__.py", line 10, in main
    return cli.run( argv[1:] )
  File "/home/kairsten/miniconda3/envs/ncov-ingest/lib/python3.6/site-packages/nextstrain/cli/__init__.py", line 56, in run
    return opts.__command__.run(opts)
  File "/home/kairsten/miniconda3/envs/ncov-ingest/lib/python3.6/site-packages/nextstrain/cli/command/build.py", line 80, in run
    return runner.run(opts, working_volume = opts.build)
  File "/home/kairsten/miniconda3/envs/ncov-ingest/lib/python3.6/site-packages/nextstrain/cli/runner/__init__.py", line 147, in run
    return opts.__runner__.run(opts, argv, working_volume = working_volume, extra_env = extra_env)
  File "/home/kairsten/miniconda3/envs/ncov-ingest/lib/python3.6/site-packages/nextstrain/cli/runner/aws_batch/__init__.py", line 114, in run
    bucket = s3.bucket(opts.s3_bucket)
  File "/home/kairsten/miniconda3/envs/ncov-ingest/lib/python3.6/site-packages/nextstrain/cli/runner/aws_batch/s3.py", line 195, in bucket
    raise ValueError('Bucket named "%s" does not exist' % bucket.name)
ValueError: Bucket named "nextstrain-jobs" does not exist
Nextstrain Run ID: cd3dad13-36bb-400e-a8aa-70c651779969
Uploading /home/kairsten/nextstrain/ncov-ingest/ncov to S3
```

Notably, the Nextstrain CLI doesn't exit with an error. I'm using version 1.16.2.
